solidity/contracts/TokenFactory.sol

```diff
@@ -0,0 +1,28 @@
+//SPDX-License-Identifier: Unlicense
+
+pragma solidity ^0.8.0;
+
+import '@openzeppelin/contracts/utils/Context.sol';
+import './ERC20WithData.sol';
+import './ERC721WithData.sol';
+
+contract TokenFactory is Context {
+    event TokenCreate(address indexed contract_address, string name, string symbol, bool is_fungible, bytes data);
+
+    function create(
+        string memory name,
+        string memory symbol,
+        bool is_fungible,
+        bytes calldata data
+    ) external virtual {
+        if (is_fungible) {
```

Comment:
I think it would be great to include some comments in this file - making a few points:

- This illustrates the factory pattern, which is a great way to deploy lots of contracts because you only have to have the blockchain overhead (gas etc.) of writing the contract code once
- The purpose is to provide a super easy to get started with sample, **not** to provide a reference implementation
- Any user going into production with an ERC-20/ERC-721 should read good practice from other communities, like OpenZeppelin, directly - rather than just relying on the FireFly community (because FF focusses on the access/indexing layer)
- This does **not** include many other good practices for blockchain smart contracts, such as upgradability (such as a proxy layer).
- This shows two points of customization that are really great `symbol` and `name`, but readers might have many other types of customization not covered here. Anything from initial supply and rules for minting/burning, through to full DAO management of the supply ecosystem.

Reply:
Yea, agreed. I think these sorts of pointers may need to exist at multiple layers (in comments, in the README, in the docs). Will circle back with a pass on some of that here.

Reply:
Added a bunch of comments along these lines. Please review and let me know if you think any changes are needed.

Reply:
This is great 👍 - thanks @awrichar